weave/type_handlers/Video/video.py

```diff
@@ -0,0 +1,135 @@
+"""Defines the custom Video weave type."""
+
+from __future__ import annotations
+
+import os
+import shutil
+from typing import TYPE_CHECKING, Any
+
+from weave.trace.serialization import serializer
+from weave.trace.serialization.custom_objs import MemTraceFilesArtifact
+
+try:
+    from moviepy.editor import (
+        VideoClip,
+        VideoFileClip,
+    )
+except ImportError:
+    dependencies_met = False
+else:
+    dependencies_met = True
+
+if TYPE_CHECKING:
+    from moviepy.editor import (
+        VideoClip,
+        VideoFileClip,
+    )
+
+SUPPORTED_FORMATS = ["gif", "mp4", "webm"]
+DEFAULT_VIDEO_FORMAT = "gif"
+
+
+def get_format_from_filename(filename: str) -> str | None:
+    """Get the file format from a filename.
+
+    Args:
+        filename: The filename to extract the format from
+
+    Returns:
+        The format string or None if no extension is found
+    """
+    # Handle special case for just a file extension (like ".mp4")
+    if filename.startswith(".") and len(filename) > 1:
+        return filename[1:]
+
+    # Use splitext which handles correctly the last extension
+    _, ext = os.path.splitext(filename)
+    if ext and len(ext) > 1:
+        return ext[1:]  # Get the extension without the dot
+    return None
+
+
+def save(
+    obj: VideoClip,
+    artifact: MemTraceFilesArtifact,
+    name: str,
+) -> None:
+    """Save a VideoClip to the artifact.
+    Args:
+        obj: The VideoClip or VideoWithPreview to save
+        artifact: The artifact to save to
+        name: Ignored, see comment below
+    """
+    is_video_file = isinstance(obj, VideoFileClip)
+
+    if is_video_file:
+        video_format = get_format_from_filename(obj.filename) or DEFAULT_VIDEO_FORMAT
+    else:
+        video_format = getattr(obj, "format", DEFAULT_VIDEO_FORMAT)
```

Comment:
I'm not sure I understand the default treatment.
According to your code, if we don't recognize the video format, we assume it is `gif`. So we will write an unknown video format to a GIF file. Do we have any guarantee that playback will work?

Reply:
The write_gif method should do the encoding with ffmpeg behind the scenes. If we think audio will often be a part of the clip we can do MP4 instead